modules/tournament2/join.php

```diff
@@ -49,29 +49,38 @@
                 $error = array();
 
                 // If joining an existing team
-                if (($_POST['existing_team_name'] != "") and ($tournament['teamplayer'] > 1)) {
-                    $success = $tteam->join($_POST["existing_team_name"], $auth["userid"], $_POST["password"]);
+                $existingTeamNameParameter = $_POST['existing_team_name'] ?? '';
+                if (($existingTeamNameParameter != "") && ($tournament['teamplayer'] > 1)) {
+                    $success = $tteam->join($existingTeamNameParameter, $auth["userid"], $_POST["password"]);
 
                 // If creating a new team
                 } else {
                     if ($tournament['teamplayer'] == 1) {
                         $_POST['team_name'] = "";
                     }
 
-                    if ($_POST["set_password"] and $_POST["set_password"] != $_POST["set_password2"]) {
+                    $setPasswordParameter = $_POST["set_password"] ?? '';
+                    $setPassword2Parameter = $_POST["set_password2"] ?? '';
+                    if ($setPasswordParameter && $setPasswordParameter != $setPassword2Parameter) {
                         $error["set_password2"] = t('Die Passworteingaben stimmen nicht überein');
                     }
                     if ($_POST['team_name'] == "" and $tournament['teamplayer'] > 1) {
                         $error["team_name"] = t('Bitte gib einen Teamnamen ein, oder wähle ein vorhandenes Team aus');
                     }
                     if (count($error) == 0) {
-                        $success = $tteam->create($_GET["tournamentid"], $auth["userid"], $_POST['team_name'], $_POST["set_password"], $_POST['team_comment'], "team_banner");
+                        $success = $tteam->create($_GET["tournamentid"], $auth["userid"], $_POST['team_name'], $setPasswordParameter, $_POST['team_comment'], "team_banner");
                     }
                 }
 
                 if (count($error) == 0 and $success) {
                     // Update-League-IDs
-                    $tteam->UpdateLeagueIDs($auth["userid"], $_POST["wwclid"], $_POST["wwclclanid"], $_POST["nglid"], $_POST["nglclanid"], $_POST["lgzid"], $_POST["lgzclanid"]);
```

Comment:
I'm reasonably certain all of these leagues are dead and all the related code can be dropped

Reply:
That is true. I do assume this would be a bigger change across the codebase. Hence, I would keep this excluded in this PR.
This is only "to get the current state somehow working properly".